### ISSUE TITLE
[Bug] fail2ban not working with roundcube

### ISSUE THREAD

**Issue by jim-perkins** [hestiacp/hestiacp#4505]
### Describe the bug

fail2ban is not enabled for roundcube and the included roundcube filter regex is not working

https://forum.hestiacp.com/t/fail2ban-doesnt-work-with-roundcube/15424/21

### Tell us how to replicate the bug

open up roundcube
attempt a bad login
notice that fail2ban is not picking it up

### Which components are affected by this bug?

Control Panel Backend, Control Panel Web Interface, Mail Security (Antivirus, Antispam), Webmail (Roundcube, SnappyMail), Firewall (Iptables, Fail2ban)

### Hestia Control Panel Version

VERSION='1.8.11'

### Operating system

debian 12

**Comment by Skamasle** [comment]
It seems that log format change a litte

I will change prefregex and failregex in roundcube-auth.conf for this

```
[Definition]

prefregex = ^\s*(\[\])?(%(__hostname)s\s*(?:roundcube(?:\[(\d*)\])?:)?\s*(<[\w\d]+>)? IMAP Error)?: <F-CONTENT>.+</F-CONTENT>$

failregex = ^(?:FAILED login|Login failed) for <F-USER>.*</F-USER>(?:against \S+)? from <HOST> (?:(?:\([^\)]*\))?\. (?:(?! from ).)*(?: user=(?P=user))? in \S+\.php on line \d+ \(\S+ \S+\))?$
            ^(?:<[\w]+> )?Failed login for <F-USER>.*</F-USER> from <HOST> in session \w+( \(error: \d\))?$
```
This not break old format

I add \d > <[\w\d]+>

And (?:against \S+)? as optional as most important change

I see It has been suggested forum in a slightly different format

I test and this works fine:

```
Results
=======

Prefregex: 18 total
|  ^\s*(\[\])?(\S+\s*(?:roundcube(?:\[(\d*)\])?:)?\s*(<[\w\d]+>)? IMAP Error)?: (?P<content>.+)$
`-

Failregex: 12 total
|-  #) [# of hits] regular expression
|   1) [12] ^(?:FAILED login|Login failed) for <F-USER>.*</F-USER>(?:against \S+)? from <HOST> (?:(?:\([^\)]*\))?\. (?:(?! from ).)*(?: user=(?P=user))? in \S+\.php on line \d+ \(\S+ \S+\))?$
`-

Ignoreregex: 0 total

Date template hits:
|- [# of hits] date format
|  [18] {^LN-BEG}Day(?P<_sep>[-/])MON(?P=_sep)ExYear[ :]?24hour:Minute:Second(?:\.Microseconds)?(?: Zone offset)?
`-

Lines: 18 lines, 0 ignored, 12 matched, 6 missed
[processed in 0.00 sec]
```

**Comment by jim-perkins** [comment]
Many  thanks. 

 So has hestia been tracking roundcube-auth.conf  ?  Since fail2ban is a debian package will the changes be overwritten with an `apt update`

and would this have been avoided if roundcube was just installed via debian.  I grant it that the hestia customizations would still need to be added in and care taken not to be overwritten by `apt update`

Hestia  is so fun to learn.  It makes things so much easier.  I havent missed forgetting to update the serial in a zone file . :-) 
However the surface is so much larger and  I have to get comfortable with that.  thanks again

**Comment by jaapmarcus** [comment]
> and would this have been avoided if roundcube was just installed via debian. I grant it that the hestia customizations would still need to be added in and care taken not to be overwritten by `apt update`
> 

In that case we would up with an older Roundcube mainly form Ubuntu 20.04 or 22.04...   Because APT updates often a lot slower ...

**Comment by Skamasle** [comment]
Problem is with roundcube 1.4 they change log format

I make a pull request to fail2ban https://github.com/fail2ban/fail2ban/pull/3816

May need report to debian mantainers also.

**Comment by jim-perkins** [comment]
Yes I can see where a good portion of the user base would like the most recent roundcube available.
Thanks for generating a pull request at fail2ban.  

Should I close this issue ?  How will the users know they need to make a change until the pull request makes it into debian/ubuntu ?

**Comment by Skamasle** [comment]
Fail2ban team fix the issue, then need some time for new version, debian
maintainers need add this to repos, so maybe months of time, I will try
report this to debian maintainers also.
